internal/http/client_test.go

```diff
@@ -0,0 +1,384 @@
+package http_test
+
+import (
+	"fmt"
+	internalhttp "github.com/newrelic/nri-discovery-kubernetes/internal/http"
+	log "github.com/sirupsen/logrus"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/rest"
+
+	"github.com/newrelic/nri-discovery-kubernetes/internal/config"
+)
+
+const (
+	healthz                = "/healthz"
+	apiProxy               = "/api/v1/nodes/test-node/proxy"
+	prometheusMetric       = "/metric"
+	kubeletMetric          = "/kubelet-metric"
+	kubeletMetricWithDelay = "/kubelet-metric-delay"
+	nodeName               = "test-node"
+	fakeTokenFile          = "./test_data/token"
+	retries                = 3
+)
+
+func TestClientCalls(t *testing.T) {
+	l := &sync.Mutex{}
+
+	s, requests := testHTTPServerWithEndpoints(t, l, []string{healthz, prometheusMetric, kubeletMetric})
+
+	k8sClient, cf, inClusterConfig, logger := getTestData(s)
+
+	kubeletClient, err := internalhttp.New(
+		internalhttp.DefaultConnector(k8sClient, cf, inClusterConfig, logger),
+		internalhttp.WithMaxRetries(retries),
+		internalhttp.WithLogger(logger),
+	)
+
+	t.Run("creation_succeeds_receiving_200", func(t *testing.T) {
+		require.NoError(t, err)
+	})
+
+	t.Run("hits_only_local_kubelet", func(t *testing.T) {
+		require.NotNil(t, requests)
+
+		l.Lock()
+		_, found := requests[healthz]
+		l.Unlock()
+		assert.True(t, found)
+	})
+
+	t.Run("hits_kubelet_metric", func(t *testing.T) {
+		r, err := kubeletClient.Get(kubeletMetric)
+		assert.NoError(t, err)
+		assert.Equal(t, r.StatusCode, http.StatusOK)
+
+		l.Lock()
+		_, found := requests[kubeletMetric]
+		l.Unlock()
+		assert.True(t, found)
```

Comment:
Maybe we do not need subtests here. Those are check for the same tests aren't them?

```suggestion
	require.NoError(t, err, "Creation succeeds receiving 200")
	
	require.NotNil(t, requests)
	t.Lock()
	assert.Len(t, requests, 1, "Hits local kubelet only")
	assert.Contains(t, requests, healthz)
	t.Unlock()
	
	r, err := kubeletClient.Get(kubeletMetric)
	assert.NoError(t, err)
	assert.Equal(t, r.StatusCode, http.StatusOK)
	assert.Contains(t, requests, kubeletMetric, "hits kubelet metric")
```

I would check the same for the rest of the client tests.

Reply:
Yes, I see the same issue there. Maybe there could be other ways to understand it, but when I see `t.Run` I expect to see different test cases (or sub-test cases) instead of different checks for the same test case.

Reply:
Solved in [#fa65105](https://github.com/newrelic/nri-discovery-kubernetes/pull/57/commits/fa65105d14598f7d5ddf2b1ce1351b5175a80106)